subsys/spm/secure_services.c

```diff
@@ -35,4 +64,26 @@ void __TZ_NONSECURE_ENTRY_FUNC spm_request_system_reboot(void)
 	spm_system_reboot();
 }
 #endif /* CONFIG_SPM_SERVICE_REBOOT */
-#endif /* CONFIG_SPM_SECURE_SERVICES */
+
+#ifdef CONFIG_SPM_SERVICE_RNG
+/** @brief Secure Service for requesting a random number from the RNG hardware
+ *
+ * Secure Entry function for allowing Non-Secure Firmware
+ * to request a random number.
+ *
+ * Note: the function will be located in a Non-Secure
+ * Callable region of the Secure Firmware Image.
+ */
+__TZ_NONSECURE_ENTRY_FUNC
+int spm_request_random_number(u8_t *output, size_t len, size_t *olen)
+{
+	int err;
+
+	if (len != MBEDTLS_ENTROPY_MAX_GATHER) {
```

Comment:
So `len` has to match this value exactly?

Reply:
Yes, this is a limitation in the backend.